### DISCUSSION TITLE
Embed a Trame container in a pre-existing Dash dashboard

### DISCUSSION THREAD

**Discussion by ghost** [Kitware/trame#192] [Q&A]
Hi everyone !

Thank you Trame-team for the awesome job you did !! 

I have a question regarding the possibility to embed a trame-based mesh visualizer and manipulator (let's say e.g. I want buttons to choose a scalar and to clip the mesh) in a pre-existing Dash dashboard.
From the documentation, it seems like the layouts for trame are mostly fullscreen ... but on the other hand, they rely on HTML containers deep down, so maybe there is a possibility to embed this HTML container in Dash ? 

I would be super grateful if somebody could comment on this :)

Best regards & take care !
Thibault

**Comment by jourdain** [comment]
Hum, this is a tricky question.

The main issue I see is that `Dash` is stateless while `trame` is stateful. So if you are using your `Dash` setup more in a stateful manner (like 1 server/process per user), then having trame running concurrently within the same Python interpreter and have some UI in Dash and some UI in trame is totally doable. By doing that, you will embed trame ui via an iframe within Dash and will have to run trame as an async task in the background like we do in Jupyter. 

HTH,

Seb
